src/scale.jl

```diff
@@ -251,45 +253,51 @@ function apply_scale_typed!(ds, field, scale::ContinuousScale)
 end
 
 # Reorder the levels of a pooled data array
-function reorder_levels(da::PooledDataArray, order::AbstractVector)
+function reorder_levels(da::IndirectArray, order::AbstractVector)
     level_values = levels(da)
     length(order) != length(level_values) &&
             error("Discrete scale order is not of the same length as the data's levels.")
     permute!(level_values, order)
-    return PooledDataArray(da, level_values)
+    return IndirectArray(da, level_values)
 end
 
-function discretize_make_pda(values::Vector, levels=nothing)
-    if levels == nothing
-        return PooledDataArray(values)
-    else
-        return PooledDataArray(convert(Vector{eltype(levels)}, values), levels)
-    end
-end
+discretize_make_pda(values::Vector)         = IndirectArray(values)
+discretize_make_pda(values::Vector, ::Void) = IndirectArray(values)
+discretize_make_pda(values::Vector, levels) =
+    IndirectArray(map(t -> findfirst(levels, t), values), levels)
 
-function discretize_make_pda(values::DataArray, levels=nothing)
-    if levels == nothing
-        return PooledDataArray(values)
-    else
-        return PooledDataArray(convert(DataArray{eltype(levels)}, values), levels)
-    end
-end
+discretize_make_pda(values::DataArray)         = IndirectArray(values)
+discretize_make_pda(values::DataArray, ::Void) = IndirectArray(values)
+discretize_make_pda(values::DataArray, levels) =
+    IndirectArray(convert(DataArray{eltype(levels)}, values), levels)
 
-function discretize_make_pda(values::Range, levels=nothing)
-    if levels == nothing
-        return PooledDataArray(collect(values))
-    else
-        return PooledDataArray(collect(values), levels)
-    end
-end
+discretize_make_pda(values::Range)         = IndirectArray(collect(values))
+discretize_make_pda(values::Range, ::Void) = IndirectArray(collect(values))
+discretize_make_pda(values::Range, levels) = IndirectArray(collect(values), levels)
 
-function discretize_make_pda(values::PooledDataArray, levels=nothing)
-    if levels == nothing
-        return values
-    else
-        return PooledDataArray(values, convert(Vector{eltype(values)}, levels))
-    end
+discretize_make_pda(values::IndirectArray)         = values
+discretize_make_pda(values::IndirectArray, ::Void) = values
+discretize_make_pda(values::IndirectArray, levels) =
+    IndirectArray(values, convert(Vector{eltype(values)}, levels))
+
+discretize_make_pda(values::CategoricalArray)         = discretize_make_pda(values, unique(values))
+discretize_make_pda(values::CategoricalArray, ::Void) = discretize_make_pda(values)
+function discretize_make_pda(values::CategoricalArray{T}, levels::Vector) where {T}
+    index = map!(t -> ismissing(t) ?
+                    findfirst(ismissing, levels) :
```

Comment:
Really weird indentation here.

Reply:
It is the first four space aligned position after the `(`. Arguments are aligned to the `(` but this splits a ternary instead of two arguments so it should be indented. The call is not super easy to read as is but the indentation makes it easier.

Reply:
AFAICT there are only three spaces, not four (assuming the count starts *after* `(`, i.e. starts with the `t`). More importantly, what bugs me is that the second operand isn't aligned with the first one.